packages/db/index.js

```diff
@@ -325,6 +325,98 @@ export class DBClient {
     }))
   }
 
+  /**
+   * List a users pinned files.
+   * @param {number} userId
+   * @param {import('./db-client-types').ListUploadsOptions} [opts]
+   * @returns {Promise<Array<import('./db-client-types').UploadItemOutput>>}
+   */
+  async listPinned (userId, opts = {}) {
+    // Query for pins the same shape as the uploads but with !inner joins
+    const pinnedQuery = `
+      _id:id::text,
+      name,
+      created:inserted_at,
+      updated:updated_at,
+      content!inner(
+        cid,
+        dagSize:dag_size,
+        pins:pin!inner(
+          status,
+          updated:updated_at,
+          location:pin_location(
+            _id:id,
+            peerId:peer_id,
+            peerName:peer_name,
+            ipfsPeerId:ipfs_peer_id,
+            region
+          )))`
+    const size = opts?.size || 10
+    // TODO get pinned for all of the users auth keys?
+    const authKeys = await this.listKeys(userId)
+    const authKeyIds = authKeys.map((authKey) => authKey._id)
+
+    let query = this._client
+      .from(psaPinRequestTableName)
+      .select(pinnedQuery)
+      .eq('content.pins.status', 'Pinned')
+      .in('auth_key_id', authKeyIds)
+      .is('deleted_at', null)
+      .limit(size)
+      .order(
+        opts.sortBy === 'Name' ? 'name' : 'inserted_at',
+        { ascending: opts.sortOrder === 'Asc' }
+      )
+
+    if (opts.before) {
+      query = query.lte('inserted_at', opts.before)
+    }
+
+    if (opts.after) {
+      query = query.gte('inserted_at', opts.after)
+    }
+
+    const { data: pins, error } = await query
+    if (error) {
+      throw new DBError(error)
+    }
+
+    const cids = pins?.map((u) => u.content.cid)
+    const deals = await this.getDealsForCids(cids)
+    return pins?.map((pin) => ({
+      ...normalizePsaPinRequest(pin),
+      type: 'PinRequest',
+      ...pin.content,
+      deals: deals[pin.content.cid] || []
+    }))
```

Comment:
Should this have `|| []` on the end, so that the return type is always an array, even if `pins?.map` is undefined?